### PR TITLE
fix(share): Don't print twice the same text on emails

### PR DESCRIPTION
**Pull request by solracsf**
## Summary

Don't print twice the same text on email.

Also removed those `» «` From `DefaultShare` as they are not printed on `ShareByMail` for consistency.

### Before

image

### After

image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/sharebymail/lib/ShareByMailProvider.php [hunk 1]
```diff
@@ -354,8 +354,7 @@ protected function sendEmail(IShare $share, array $emails): void {
 		}
 
 		$emailTemplate->addBodyText(
-			htmlspecialchars($text . ' ' . $this->l->t('Click the button below to open it.')),
-			$text
+			$this->l->t('Click the button below to open it.')
```
Comment (come-nc) [inline apps/sharebymail/lib/ShareByMailProvider.php:357]:
```suggestion
			htmlspecialchars($this->l->t('Click the button below to open it.'))
```
The translation may contain special characters depending on the language, so you should still escape them.

Comment (come-nc) [reply apps/sharebymail/lib/ShareByMailProvider.php:357]: Also I think `$text` is now unused and show be removed?

**Review by artonge — APPROVED** [review]

File: apps/sharebymail/lib/ShareByMailProvider.php (hunk 1 above)
Comment (solracsf) [reply apps/sharebymail/lib/ShareByMailProvider.php:357]: > The translation may contain special characters depending on the language, so you should still escape them.

Hum...but all translations are unescaped, unless they have direct user input (like `$text` or `$note`)...are you absolutely sure here?

Comment (solracsf) [reply apps/sharebymail/lib/ShareByMailProvider.php:356]: > Also I think `$text` is now unused and show be removed?

Fixed 👍

File: lib/private/Share20/DefaultShareProvider.php [hunk 2]
```diff
@@ -1512,21 +1512,21 @@ protected function sendUserShareMail(
 			'shareWith' => $shareWith,
 		]);
 
-		$emailTemplate->setSubject($l->t('%1$s shared »%2$s« with you', [$initiatorDisplayName, $filename]));
+		$emailTemplate->setSubject($l->t('%1$s shared %2$s with you', [$initiatorDisplayName, $filename]));
 		$emailTemplate->addHeader();
-		$emailTemplate->addHeading($l->t('%1$s shared »%2$s« with you', [$initiatorDisplayName, $filename]), false);
-		$text = $l->t('%1$s shared »%2$s« with you.', [$initiatorDisplayName, $filename]);
+		$emailTemplate->addHeading($l->t('%1$s shared %2$s with you', [$initiatorDisplayName, $filename]), false);
+		$text = $l->t('%1$s shared %2$s with you.', [$initiatorDisplayName, $filename]);
```
[the remark this answers is not in the document]
Comment (solracsf) [reply lib/private/Share20/DefaultShareProvider.php]: Arf... forgot this one 😆

**Comment by solracsf** [comment]
CI seems broken because of https://github.com/nextcloud/server/pull/49693 😢

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Rebased, cypsress passes now

**Review by jancborchardt — COMMENTED** [review]
Nice update! :) I would also suggest to remove the text "Click the button below to open it" as it’s a duplication of the button right below, and just creates more visual noise.

@solracsf in case you also would like to submit a pull request for that as it improves it even more? :)

**Pull request merged**